### PR TITLE
feat: Added histograms for ctag fixed WP SF

### PR DESCRIPTION
**Pull request by adeiorio**
util: add WP hist
wf: add WP hists & selections

### CODE REVIEW

File: src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py [hunk 1]
```diff
@@ -626,6 +647,17 @@ def process_shift(self, events, shift_name):
                     ak.values_astype(events[event_level].Pileup.nTrueInt, int),
                     weight=weight,
                 )
+            if "Wc" in self.selMod:
```
Comment (Ming-Yan) [inline src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:650]: Could you please add a comment this is for WP-based SF? Also, from the `if` statement here, it seems to me you want to add the WP histogram for all the W+c workflow including shape based, did I understand correctly?

File: src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py [hunk 2]
```diff
@@ -102,13 +104,18 @@ def process_shift(self, events, shift_name):
...
             triggers = ["IsoMu27", "IsoMu24"]
             isMu = True
             dxySigcut = 1.0
             muNeEmSum = 0.7
             muonpTratioCut = 0.4
             isolepdz, isolepdxy, isolepsip3d = 0.01, 0.002, 2
+        ### remove muNeEmSum for cutbased
+        if "cutbased_WcM" == self.selMod:
+            muNeEmSum = 1.0
+            muonpTratioCut = 0.6
```
Comment (Ming-Yan) [inline src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: @mondalspandan can we sync the selection? 
@adeiorio we see a problem with higher pT threshold (was 0.75 [before for shape](https://indico.cern.ch/event/1339673/contributions/5640056/attachments/2739110/4764230/BTV_commissioning_231023.pdf), and we observed some issue. Do you observed good data/MC agreement?

**Review by Ming-Yan — COMMENTED** [review]
Hi @adeiorio thank you very much for your PR and I have some question.
The changes looks all good to me.
As you see the lining is failed, can you please run `black` in your local repo and recommit again? we are using version `24.8.0`

File: src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py (hunk 1 above)
Comment (adeiorio) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:653]: For fixed WP-based SFs these histograms are used for the analysis. We could also remove from shape analysis.

Comment (Ming-Yan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:653]: I think either way is fine :p, just not used in the shape based. I want to cross check

File: src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py (hunk 2 above)
Comment (mondalspandan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: IIRC the issue was with the `muNeEmSum` cut not being well calibrated. It kills off a lot of DY but can introduce an additional norm change. I am happy to remove this cut.

Comment (Ming-Yan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: Hi @adeiorio @mondalspandan thanks, so both energy fraction cut should in sync?
Then which one we should take?

Comment (mondalspandan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: Let's keep `muNeEmSum = 1.0` since it's the safest.

Comment (Ming-Yan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: How about muonpTratioCut ? @mondalspandan

Comment (Ming-Yan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: So we should leave `muonpTratioCut = 0.4 and muNeEmSum = 1.0` @mondalspandan @adeiorio  are you agree on this?

Comment (adeiorio) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: Hi @Ming-Yan I produced a new round of data-MC for 2023 with this selection and results are [here](https://adeiorio.web.cern.ch/adeiorio/BTV/Summer23/v2/ctag_Wc_WP_sf__24_10_16/)

Comment (Ming-Yan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: This looks ok to me, I think we should merge this 
@adeiorio @mondalspandan what do you think?

Comment (mondalspandan) [reply src/BTVNanoCommissioning/workflows/ctag_Wctt_valid_sf.py:118]: It's of course ok with me :)

**Review by Ming-Yan — APPROVED** [review]
Thanks for changing this and also the studies on harmonize the selections. Will merge it!

**Pull request merged**